streams/src/main/java/org/apache/kafka/streams/kstream/CogroupedKStream.java

```diff
@@ -0,0 +1,295 @@
+ * KGroupedStream#cogroup(Aggregator) cogroup(...)}.
+ *
+ * @param <K> Type of keys
+ * @param <VOut> Type of values after agg
+ */
+public interface CogroupedKStream<K, VOut> {
+
+    /**
+     * Add an already {@link KGroupedStream grouped KStream} to this {@code CoGroupedKStream}.
+     * {@code CogroupedKStream} is an abstraction of multiple <i>grouped</i> record streams of
+     * {@link KeyValue} pairs.
+     * It is an intermediate representation of one or more {@link KStream}s
+     * in order to apply one or more aggregation operations on the original {@link KStream}
+     * records.
+     * <p>
+     * The specified {@link Aggregator} is applied for each input record and computes
+     * a new aggregate using the current aggregate (or for the very first record using the
+     * intermediate aggregation result provided via the {@link Initializer}) and the record's value.
+     * <p>
+     * Thus, {@code aggregate(Initializer, Materialized)} can be used to compute
+     * aggregate functions like count.
+     * It is an intermediate representation after a grouping of {@link KStream}s, before the
+     * aggregations are applied to the new partitions resulting in a {@link KTable}.
+     * @param groupedStream a group stream object
+     * @param aggregator   an {@link Aggregator} that computes a new aggregate result
+     * @param <VIn> Type of input values
+     * @return a {@code CogroupedKStream}
+     */
+    <VIn> CogroupedKStream<K, VOut> cogroup(final KGroupedStream<K, VIn> groupedStream,
+                                            final Aggregator<? super K, ? super VIn, VOut> aggregator);
+
+    /**
+     * Aggregate the values of records in these streams by the grouped key.
+     * Records with {@code null} key or value are ignored.
+     * The result is written into a local {@link KeyValueStore} (which is basically an
+     * ever-updating materialized view) that can be queried by the given store name in {@code
+     * materialized}.
+     * Furthermore, updates to the store are sent downstream into a {@link KTable} changelog stream.
+     * <p>
+     * The specified {@link Initializer} is applied once directly before the first input record is
+     * processed to provide an initial intermediate aggregation result that is used to process the
+     * first record.
+     * Not all updates might get sent downstream, as an internal cache is used to deduplicate
+     * consecutive updates to the same key. The rate of propagated updates depends on your input
+     * data rate, the number of distinct keys, the number of parallel running Kafka Streams
+     * instances, and the {@link StreamsConfig configuration} parameters for {@link
+     * StreamsConfig#CACHE_MAX_BYTES_BUFFERING_CONFIG cache size}, and {@link
+     * StreamsConfig#COMMIT_INTERVAL_MS_CONFIG commit intervall}.
+     * <p>
+     * To query the local {@link KeyValueStore} it must be obtained via {@link
```

Comment:
It does not require a TimestampedKeyValueStore

Reply:
https://issues.apache.org/jira/browse/KAFKA-9243

Reply:
> It does not require a TimestampedKeyValueStore

`Materialized` does take `KeyValueStore<Bytes,byte[]>` but this byte-arrya-kv-store was wrapped by `KeyValueStore<K,V>` before but is now wrapped by a `TimestampedKeyValueStore<K,V>` -- it's not about the API of the method, but about documenting IQ correctly.

But we can also leave the JavaDocs as-is for now, as the show code will still work. Thanks for creating a ticket to fix all JavaDocs at once (this way, it will be at least consistent).